torchtext/datasets/qnli.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import os
+import csv
+
+from torchtext._internal.module_utils import is_module_available
+from torchtext.data.datasets_utils import (
+    _create_dataset_directory,
+    _wrap_split_argument,
+)
+
+if is_module_available("torchdata"):
+    from torchdata.datapipes.iter import FileOpener, IterableWrapper
+
+    # we import HttpReader from _download_hooks so we can swap out public URLs
+    # with interal URLs when the dataset is used within Facebook
+    from torchtext._download_hooks import HttpReader
+
+
+URL = "https://dl.fbaipublicfiles.com/glue/data/QNLIv2.zip"
+
+MD5 = "b4efd6554440de1712e9b54e14760e82"
+
+NUM_LINES = {
+    "train": 104743,
+    "dev": 5463,
+    "test": 5463,
+}
+
+_PATH = "QNLIv2.zip"
+
+DATASET_NAME = "QNLI"
+
+_EXTRACTED_FILES = {
+    "train": os.path.join("QNLI", "train.tsv"),
+    "dev": os.path.join("QNLI", "dev.tsv"),
+    "test": os.path.join("QNLI", "test.tsv"),
+}
+
+
+@_create_dataset_directory(dataset_name=DATASET_NAME)
+@_wrap_split_argument(("train", "dev", "test"))
+def QNLI(root, split):
+    """QNLI Dataset
+
+    For additional details refer to https://arxiv.org/pdf/1804.07461.pdf (from GLUE paper)
+
+    Number of lines per split:
+        - train: 104743
+        - dev: 5463
+        - test: 5463
+
+    Args:
+        root: Directory where the datasets are saved. Default: os.path.expanduser('~/.torchtext/cache')
+        split: split or splits to be returned. Can be a string or tuple of strings. Default: (`train`, `dev`, `test`)
+
+    :returns: DataPipe that yields tuple of text and label (0 and 1).
+    :rtype: (int, str, str)
+    """
+    # TODO Remove this after removing conditional dependency
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError(
+            "Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`"
+        )
+
+    def _filepath_fn(x=None):
+        return os.path.join(root, os.path.basename(x))
+
+    def _extracted_filepath_fn(_=None):
+        return os.path.join(root, _EXTRACTED_FILES[split])
+
+    def _filter_fn(x):
+        return _EXTRACTED_FILES[split] in x[0]
+
+    def _modify_res(x):
+        return (int(x[3]=="entailment"), x[1], x[2])
+
+
+    url_dp = IterableWrapper([URL])
+    cache_compressed_dp = url_dp.on_disk_cache(
+        filepath_fn=_filepath_fn,
+        hash_dict={_filepath_fn(URL): MD5},
+        hash_type="md5",
+    )
+    cache_compressed_dp = HttpReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(
+        filepath_fn=_extracted_filepath_fn
+    )
+    cache_decompressed_dp = (
+        FileOpener(cache_decompressed_dp, mode="b").read_from_zip().filter(_filter_fn)
```

Comment:
Can we change this to `load_from_zip`? As per the [warning message in torchdata](https://github.com/pytorch/data/blob/c1d89fe9a1b06e610f32f823359771557b1ca12a/torchdata/datapipes/iter/util/ziparchiveloader.py#L89-L96), `read_from_zip` is deprecated

Reply:
Thanks for catching this, done.